src/java/org/apache/cassandra/utils/vint/VIntCoding.java

```diff
@@ -165,6 +190,27 @@ public static long readVInt(DataInput input) throws IOException
         return decodeZigZag64(readUnsignedVInt(input));
     }
 
+    /**
+     * Read up to a signed 32-bit integer back, throwing IllegalArgumentException if
+     * the vint read is out of range for a 32-bit integer
+     * @throws IllegalArgumentException Thrown if the vint doesn't fit a 32-bit integer
+     */
+    public static int readVInt32(DataInput input) throws IOException
+    {
+        return checkedCast(decodeZigZag64(readUnsignedVInt(input)));
+    }
+
+    /**
+     * Read up to a 32-bit unsigned integer. This uses a more efficient encoding
+     * for numbers that are mostly positive (avoids zigzag).
+     *
+     * @throws IllegalArgumentException Thrown if the vint doesn't fit a 32-bit integer
```

Comment:
Should this be `VIntOutOfRangeException` now instead of `IllegalArgumentException`?

Reply:
super-nit: `fit a` -> `fit into a`?

Reply:
Changed it